src/extension.ts

```diff
@@ -244,7 +244,7 @@ export async function activate(context: vscode.ExtensionContext) {
 
         recordToolkitInitialization(activationStartedOn, getLogger())
 
-        assertPassiveTelemetry()
+        ext.telemetry.assertPassiveTelemetry(ext.didReload())
```

Comment:
I was being bitten by this bug when messing around with another branch with the last change and noticed that this didn't block extension load (since this is at the end of the activation) yet I still got the error toast. Since this is just for development purposes only, would it be worth suppressing the error message in case of this telemetry error? That way, users won't know a thing and CI will still break.

Additionally, should this affect the `recordToolkitInitialization` reported status?

Reply:
good idea! But I think we want users to report this bug. The log message will tell us which metric was emitted. Also, we want to see the issue in development. CI can't cover all possible telemetry permutations.

I will look at `recordToolkitInitialization`.

Reply:
I don't think there are any other adverse effects, so maybe pop a different message? That being said, if we want to see failures as they happen, a change to the metrics might be able to cover that.

Reply:
Now it fails silently if the Toolkit version is a release build like `1.22.0` (as opposed to `1.22.0-foo`).